db/models.js

```diff
@@ -1,7 +1,23 @@
 const db = require("./connection");
+const fs = require("fs/promises");
 
 exports.selectTopics = () => {
   return db.query("SELECT * FROM topics").then((topicsData) => {
     return topicsData.rows;
   });
 };
+
+exports.readDataFile = () => {
+  const filePath = `${__dirname}/../endpoints.json`;
+
+  return fs
+    .readFile(filePath, "utf8")
+    .then((fileData) => {
+      return JSON.parse(fileData);
+    })
+    .catch((err) => {
+      if (err.code === "ENOENT") {
+        throw new Error("File not found");
+      }
+    });
```

Comment:
We don't need this catch block here if we have a catch on our controller as it will catch the error from the model.